support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -750,6 +777,66 @@ export function Checkout() {
 											/>
 										</>
 									)}
+
+									{paymentMethod === 'DirectDebit' && (
+										<DirectDebitForm
+											countryGroupId={countryGroupId}
+											accountHolderName={accountHolderName}
+											accountNumber={accountNumber}
+											accountHolderConfirmation={accountHolderConfirmation}
+											sortCode={sortCode}
+											recaptchaCompleted={false}
+											updateAccountHolderName={(name: string) => {
+												setAccountHolderName(name);
+											}}
+											updateAccountNumber={(number: string) => {
+												setAccountNumber(number);
+											}}
+											updateSortCode={(sortCode: string) => {
+												setSortCode(sortCode);
+											}}
+											updateAccountHolderConfirmation={(
+												confirmation: boolean,
+											) => {
+												setAccountHolderConfirmation(confirmation);
+											}}
+											recaptcha={
+												<Recaptcha
+													// We could change the parents type to Promise and uses await here, but that has
+													// a lot of refactoring with not too much gain
+													onRecaptchaCompleted={(token) => {
+														setRecaptchaToken(token);
+														void fetch(
```

Comment:
I don't think we need to make this `fetch` in the `onRecaptchaCompleted` function for Direct Debit, this is only necessary for Credit/Debit card (eg. w/ Stripe)

Reply:
<img width="679" alt="Screenshot 2024-03-28 at 09 56 53" src="https://github.com/guardian/support-frontend/assets/31692/bf47a757-8d8d-42eb-b0a7-5890c56e6f00">

When the post is made in `PROD` - this value is set, so I am just replicating that. [This is validated on the server](https://github.com/guardian/support-frontend/blob/main/support-frontend/app/controllers/CreateSubscriptionController.scala#L81-L87), so we do need to send it across.

Reply:
Do we make a `POST` to `/stripe/create-setup-intent/recaptcha` in the `onRecaptchaCompleted` for the Direct Debit form? I can't see that in the network requests, I only see it when checking the Recpatcha in the Credit/Debit card section of the form.

For the current Recaptcha in the Direct Debit section of the form we define `onRecaptchaCompleted` [here](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/components/directDebit/directDebitForm/directDebitFormContainer.tsx#L65), this in turn calls the Recaptcha [reducer](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/redux/checkout/recaptcha/reducer.ts#L12) which sets the captcha token in state, this is then sent to the server in the `/create` payload for validation.

Reply:
Got you - I've added the second step of using the `recaptchToken` to get a `stripe-intent`, which is a silly copy/pasta.

This has been removed.